azurelinuxagent/ga/exthandlers.py

```diff
@@ -783,6 +783,19 @@ def __init__(self, ext_handler, protocol):
             self.logger.error(u"Failed to create extension log dir: {0}", e)
 
         log_file = os.path.join(self.get_log_dir(), "CommandExecution.log")
+
+        try:
+            if os.stat(log_file).st_size > execution_log_max_size:
+                with open(log_file, "rb") as existing_log_file_buffer:
+                    existing_log_file_buffer.seek(-1 * execution_log_max_size, 2)
+                    _ = existing_log_file_buffer.readline()
+
+                    fileutil.write_file("".join([log_file, ".tmp"]), existing_log_file_buffer.read(), asbin=True)
+                    os.rename("".join([log_file, ".tmp"]), log_file)
+
+        except (IOError, OSError):
+            pass
```

Comment:
Incase of an error should we consider just deleting the file rather than keep on adding to it?

Reply:
The purpose of this except line was to catch a `FileNotFound` error thrown by `os.stat`; `log_file` is not necessarily already there, and "better ask forgiveness than permission" is considered pythonic.

There're more file operations within this try block, though, so it might be worth handling those separately...

Reply:
Why not just do a blanket catch and delete file if any random error? If considering the `FileNotFound` error also, deleting a file that's not there is just going to be a no-op

Reply:
to be clear I only mean delete the file if the size > MAX_SIZE and we throw some error when trying to truncate it

Reply:
` "better ask forgiveness than permission" is considered pythonic.
`
Sure, but let's ask for forgiveness explicitly, catch FileNotFound and handle it appropriately

Reply:
Unfortunately, I believe FileNotFound is python3 only. I originally had that as the exception name.

Reply:
> I'm leaning towards writing a function in future.py that determines if an IOError is from a file not found exception or not.

Sounds like a good idea